overlord/restart/restart.go

```diff
@@ -117,6 +122,57 @@ func (m *RestartManager) Ensure() error {
 	return nil
 }
 
+// StartUp implements StateStarterUp.Startup.
+func (m *RestartManager) StartUp() error {
+	s := m.state
+	s.Lock()
+	defer s.Unlock()
+
+	// mark as done tasks that were waiting for an external reboot
+	for _, chg := range s.Changes() {
+		if chg.IsReady() {
+			continue
+		}
+		heldCount, err := changeHeldForSystemRestart(chg)
+		if err != nil {
+			return err
+		}
+		if heldCount == 0 {
+			continue
+		}
+		for _, t := range chg.Tasks() {
```

Comment:
This check could be moved inside the inner loop so that we exit early after processing the tasks held for the restart. It would also ensure that we run `chg.Set("held-for-system-restart", nil)` to clean up, even if the count was zero at the start.
```suggestion
		for _, t := range chg.Tasks() {
			if heldCount == 0 {
			    break
		    }
```

Reply:
it was resolved a bit different once I turned the counter into just a flag